### PR TITLE
kafka: Enable TCP Keepalive on proxy dialed connections

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/proxy/dialer.go [hunk 1]
```diff
@@ -65,5 +80,10 @@ func ciliumDialer(identity int, network, address string) (net.Conn, error) {
 		return nil, fmt.Errorf("unable to connect: %s", err)
 	}
 
+	if err := setKeepAlive(c); err != nil {
+		c.Close()
```
Comment (manalibhutiyani) [inline pkg/proxy/dialer.go:84]: Would it make more sense to add `defer c.Close()` on line and 61 instead? and cleanup all other `c.Close` too

File: pkg/proxy/dialer.go [hunk 2]
```diff
@@ -65,5 +80,10 @@ func ciliumDialer(identity int, network, address string) (net.Conn, error) {
 		return nil, fmt.Errorf("unable to connect: %s", err)
 	}
 
+	if err := setKeepAlive(c); err != nil {
```
Comment (rlenglet) [inline pkg/proxy/dialer.go:83]: Shouldn't we also set the SO_LINGER option, like for the request connections?
```
	if err = setLinger(c, proxyConnectionCloseTimeout); err != nil {
 		c.Close()
 		return nil, err
 	}
```

**Review by rlenglet — APPROVED** [review]

File: pkg/proxy/dialer.go (hunk 1 above)
Comment (tgraf) [reply pkg/proxy/dialer.go:84]: We only want to close when the operation failed

Comment (manalibhutiyani) [reply pkg/proxy/dialer.go:84]: got it ..thanks!

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**